watcher/handlers/metalogger.go

```diff
@@ -122,18 +144,145 @@ func (l *MetaLogger) OnAdd(obj interface{}) error {
 			"__kf_workspace__": mlpbStringValue(workspace),
 		},
 	}
-	request := storepb.PutArtifactsRequest{
-		Artifacts: []*mlpb.Artifact{&artifact},
+	request := storepb.PutExecutionsRequest{
+		Executions: []*mlpb.Execution{execution},
 	}
-	resp, err := l.kfmdClient.PutArtifacts(context.Background(), &request)
+	resp, err := l.kfmdClient.PutExecutions(context.Background(), &request)
 	if err != nil {
-		klog.Errorf("failed to log metadata for %s: err = %s, request = %v, resp = %v", l.MetadataArtifactType(), err, request, resp)
+		klog.Errorf("failed to log metadata for %s: err = %s, request = %v, resp = %v", k8smlmd.GVKToType(l.resource), err, request, resp)
 		return err
 	}
-	klog.Infof("Handled addEvent for %s.\n", l.MetadataArtifactType())
+
+	if len(resp.ExecutionIds) == 0 {
+		err := fmt.Errorf("No execution id returned from MLMD")
+		log.Error(err, "Newly created execution is missing an id")
+		return err
+	}
+
+	for  _, annotation := range []string{v1alpha1.InputAnnotation, v1alpha1.OutputAnnotation} {
+		ids, err := l.logArtifactAnnotation(object, annotation)
+
+		if err != nil {
+			log.Error(err, "Could not log artifact annotation", "annotation", annotation)
+			continue
+		}
+
+		// Create events connecting the execution and artifacts
+		for _, aId := range ids {
+			// TODO(jlewi): What should we set for the path if anythong?
+			e := &mlpb.Event{
```

Comment:
I remember Event has a path field but have not tried this before. Do you know if https://github.com/google/ml-metadata/blob/127cb87df7b3509140e321c401c77a9d0a8a223c/ml_metadata/proto/metadata_store.proto#L165-L174 this can be used here?

Reply:
I don't know what the path field is supposed to be used for.  I'm guessing its some way of distinguishing among the different inputs to an execution. e.g. if you have N values and the execution takes N inputs you need some way to match the values to input; I'm guessing that's what path is for.